app/src/main/java/org/fedorahosted/freeotp/edit/EditActivity.java

```diff
@@ -138,7 +160,7 @@ public void afterTextChanged(Editable s) {
     public void onClick(View v) {
         switch (v.getId()) {
             case R.id.image:
-                startActivityForResult(new Intent(Intent.ACTION_PICK,
+                startActivityForResult(new Intent(Intent.ACTION_OPEN_DOCUMENT,
```

Comment:
To support ACTION_OPEN_DOCUMENT the minSdkVersion needs to be **increased to at least 19.**
Current min is 14

Reply:
The user can pick any file, so the app will crash if a non-image is chosen. To fix this add this type filter
from https://developer.android.com/guide/components/intents-common.html#OpenFile
<pre>
private final int REQUEST_IMAGE_OPEN = 0;

...

intent.setType("image/*");
intent.addCategory(Intent.CATEGORY_OPENABLE);
startActivityForResult(intent, REQUEST_IMAGE_OPEN);
</pre>